c8d/prune: Handle filters, don't delete used images

vvoland opened the pull request:
Support the same filters as the non-c8d docker for `image prune -a`.
Also no longer remove the content from ContentStore ourselves - let c8d do that when deleting Image with SynchronousDelete (triggers GC immediately), and check afterwards which blobs are no longer in the store.

Not sure if there's anything meaningful from our side to do in `docker image prune`, as there is no dangling images in c8d due to GC. So currently the only sensible thing is to just trigger the GC, in case there were performed some actions without `SynchronousDelete` option.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

rumpl commented on daemon/containerd/image_prune.go:
```diff
@@ -3,80 +3,172 @@ package containerd
+		leases := i.client.LeasesService()
+		lease, err := leases.Create(ctx)
+		if err != nil {
+			return &report, nil
```
We don't want to return the error here?

vvoland replied:
We do :) Thanks!

vvoland replied:
We do, thanks!

rumpl commented on daemon/containerd/image_prune.go:
```diff
@@ -3,80 +3,172 @@ package containerd
+		// So running prune with dangling=true is mostly a no-op, unless there
+		// was some action performed which didn't invoke the GC immediately.
+
+		// Trigger GC.
```
Is this the only way to trigger the gc?

vvoland replied:
I haven't seen an easier way. To invoke it directly we'd have to get to the GC/scheduler plugin and call the `(*gcScheduler).ScheduleAndWait`. But it's hacky.

vvoland replied:
I haven't seen an easier way. To invoke it directly we'd have to get to the GC/scheduler plugin and call the `(*gcScheduler).ScheduleAndWait`. But it's hacky.

vvoland commented on daemon/containerd/image_prune.go:
```diff
@@ -3,80 +3,172 @@ package containerd
+		if err != nil {
+			return &report, nil
+		}
+		leases.Delete(ctx, lease)
```
Forgot an SynchronousDelete here, without it the GC won't get triggered immediately! Fixed.

vvoland commented:
I removed the `checkPresentPlatforms` that I brought from other PR because it was actually a convoluted way to perform a recursive walk on the blobs that are present in the Store.
Changed it to a simpler ImageHandler that actually does the correct thing.

thaJeztah commented:
Should be combined when upstreaming:

- https://github.com/rumpl/moby/pull/20
